textalyze.js

```diff
@@ -17,7 +19,69 @@ function itemCounts(array) {
   // Your code here.
   // Run 'npm test' to see what tests need to pass.
 
+  array.forEach((value) => {
+    const sum = counts.get(value) ? counts.get(value) + 1 : 1;
+    counts.set(value, sum);
+  });
+
   return counts;
 }
 
-module.exports = { itemCounts };
+/**
+ * Takes an arbitrary String as input and returns an Array of all the characters in the string
+ * @param {String}
+ * @returns {Array}
+ */
+function stringCharsToArray(string) {
+  const arrayString = string.split('');
+  return itemCounts(arrayString);
+}
+
+/**
+ * Takes an arbitrary String as input and returns a "sanitized" string
+ * that replaces all upper-case letters with their lower-case equivalent
+ * @param {String}
+ * @returns {Array}
+ */
+function sanitize(string) {
+  return string.toLowerCase();
+}
+
+/**
+ * Takes an string containing the path to a file and returns the data
+ * @param {String} path to the file
+ * @returns {String} file data
+ */
+function readFromFile(file) {
+  return new Promise((resolve, reject) => {
+    fs.readFile(file, (err, data) => {
+      if (err) {
+        return reject(new Error(err));
+      }
+      return resolve(data.toString());
+    });
+  });
+}
+
+/**
+ * Given an input Array, returns a Map containing the Basic Frequency Statistics
+ * @param {Array} array of strings
+ * @returns {Map} map with frequencies
+ */
+function frequencyStatics(array) {
+  const counts = itemCounts(array);
+  const statistics = new Map();
+  let total = 0;
+
+  counts.forEach((value, index) => { total += counts.get(index); });
```

Comment:
A few things:

There's no reason to call `counts.get(index)` since `value === counts.get(index)`.

This pattern should call reduce to mind immediately:

```javascript
const add = (x, y) => x + y;
const total = [...counts.values()].reduce(add);
```

But even that is overly complicated because `array.length` is the total you want, so:

```javascript
const total = array.length;
```

Reply:
By pattern I mean:

```javascript
let total = 0;

numbers.forEach(number => total += number);

return total;
```

Any code that has that "shape" can be replaced with:

```javascript
return numbers.reduce((total, number) => total + number);
```

Reply:
Yes I thought about reduce immediately, although this is a Map iterator, and they don't have the reduce API on it. I should have thought about doing it this way `[...counts.values()]` though, it's better indeed. I did it kinda asleep, sorry